lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -279,9 +279,12 @@ public function put(QueryCacheKey $key, ResultSetMapping $rsm, $result, array $h
 
         $region = $persister->getCacheRegion();
 
+        $cm             = $this->em->getClassMetadata($entityName);
+        $rootEntityName = $cm->getMetadataValue('rootEntityName');
```

Comment:
```suggestion
```

You should directly use `$cm->rootEntityName` in line 287, `getMetadataValue` has a different use-case.

Reply:
Hi @beberlei , I originally did that but then the codesniffer job failed due to accessing a property when presented only with an interface - should we suppress or ignore that?

Reply:
@peterkeatingie yes, you can add `assert($cm instanceof ClassMetadata);` in line 283, this code gets compiled away in production when assertions are off.